include/realizations/catchment/Formulation_Manager.hpp

```diff
@@ -285,16 +291,32 @@ namespace realization {
              * 
              * @return std::string of the output root directory
              */
-            std::string get_output_root() const noexcept {
+            std::string get_output_root() const {
                 const auto output_root = this->tree.get_optional<std::string>("output_root");
                 if (output_root != boost::none && *output_root != "") {
                     // Check if the path ends with a trailing slash,
                     // otherwise add it.
-                    return output_root->back() == '/'
+                    std::string str = output_root->back() == '/'
                            ? *output_root
                            : *output_root + "/";
-                }
 
+                    const char* dir = str.c_str();
+
+                    //use C++ system function to check if there is a dir match that defined in realization
+                    struct stat sb;
+                    if (stat(dir, &sb) == 0 && S_ISDIR(sb.st_mode))
+                        return dir;
```

Comment:
```suggestion
                        return str;
```
We've already made the string. There's no reason to re-construct a copy like this.

Reply:
Yeah. It is also more consistent with the function definition.